File: miniprogram/models/AIModelConfig.js
/**
 * AI模型配置实体类
 */
class AIModelConfig {
  /**
   * 构造函数
   * @param {Object} options - 初始化选项
   * @param {Number} options.id - 模型ID
   * @param {String} options.name - 模型名称
   * @param {String} options.provider - 提供商
   * @param {String} options.model - 模型标识
   * @param {Boolean} options.isDefault - 是否为默认模型
   * @param {String} options.description - 模型描述
   * @param {String} options.apiUrl - API地址（敏感信息，前端不显示）
   * @param {String} options.apiKey - API密钥（敏感信息，前端不显示）
   * @param {Number} options.timeout - 超时时间（毫秒）
   * @param {Number} options.tokensLimit - 支持的最大Token数
   * @param {Number} options.pointsCost - 使用该模型消耗的积分数
   * @param {String} options.icon - 模型图标URL
   * @param {Boolean} options.available - 模型是否可用
   */
  constructor(options = {}) {
    this.id = options.id || null;
    this.name = options.name || '';
    this.provider = options.provider || '';
    this.model = options.model || '';
    this.isDefault = options.isDefault || false;
    this.description = options.description || '';
    this.timeout = options.timeout || 60000;
    this.tokensLimit = options.tokensLimit || 4000;
    this.pointsCost = options.pointsCost || 1;
    this.icon = options.icon || '';
    this.available = options.available !== false;
    
    // 敏感信息，仅在服务端使用
    this._apiUrl = options.apiUrl;
    this._apiKey = options.apiKey;
  }

  /**
   * 从API响应创建实例
   * @param {Object} data - API返回的数据
   * @returns {AIModelConfig} 创建的实例
   */
  static fromApiResponse(data) {
    return new AIModelConfig({
      id: data.id,
      name: data.name,
      provider: data.provider,
      model: data.model,
      isDefault: data.isDefault,
      description: data.description,
      timeout: data.timeout,
      tokensLimit: data.tokensLimit,
      pointsCost: data.pointsCost,
      icon: data.icon,
      available: data.available
    });
  }

  /**
   * 获取模型详细信息（用于展示）
   * @returns {Object} 模型详细信息
   */
  getDisplayInfo() {
    return {
      id: this.id,
      name: this.name,
      provider: this.provider,
      description: this.description,
      pointsCost: this.pointsCost,
      isDefault: this.isDefault,
      icon: this.icon
    };
  }

  /**
   * 检查模型是否可用
   * @returns {Boolean} 是否可用
   */
  isAvailable() {
    return this.available;
  }

  /**
   * 获取使用该模型所需的积分
   * @returns {Number} 所需积分
   */
  getPointsCost() {
    return this.pointsCost;
  }
}

module.exports = AIModelConfig; 

File: src/main/java/com/wordcheck/model/AIModelConfig.java
package com.wordcheck.model;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * AI模型配置实体类
 */
@Data
public class AIModelConfig {
    /**
     * 配置ID
     */
    private Integer id;
    
    /**
     * 模型名称
     */
    private String name;
    
    /**
     * 模型提供商 (如OpenAI, Azure, 百度等)
     */
    private String provider;
    
    /**
     * 模型API地址
     */
    private String apiUrl;
    
    /**
     * 具体模型标识
     */
    private String modelId;
    
    /**
     * API密钥
     */
    private String apiKey;
    
    /**
     * 是否为默认模型
     */
    private Boolean isDefault;
    
    /**
     * 超时时间(毫秒)
     */
    private Integer timeout;
    
    /**
     * 提示词模板（批改作文的指令）
     */
    private String promptTemplate;
    
    /**
     * 基础积分消耗（不足字数阈值时收取的积分）
     */
    private Integer basePointsCost;
    
    /**
     * 字数阈值（达到该字数时开始按阶梯计费）
     */
    private Integer wordThreshold;
    
    /**
     * 字数增量（每增加多少字进行一次计费）
     */
    private Integer wordIncrement;
    
    /**
     * 增量积分消耗（每增加一个wordIncrement消耗的积分）
     */
    private Integer incrementPointsCost;
    
    /**
     * 积分计费方式：0=固定积分，1=按字数梯度
     */
    private Integer pointsCostType;
    
    /**
     * 创建时间
     */
    private LocalDateTime createdAt;
    
    /**
     * 更新时间
     */
    private LocalDateTime updatedAt;
    
    /**
     * 计算指定内容长度需要消耗的积分
     * @param contentLength 内容长度
     * @return 需要消耗的积分
     */
    public int calculatePointsCost(int contentLength) {
        // 固定积分收费
        if (pointsCostType == null || pointsCostType == 0 || basePointsCost == null) {
            return basePointsCost != null ? basePointsCost : 1;
        }
        
        // 按字数梯度收费
        if (wordThreshold == null || wordThreshold <= 0) {
            wordThreshold = 1000; // 默认1000字
        }
        
        if (wordIncrement == null || wordIncrement <= 0) {
            wordIncrement = 1000; // 默认每1000字增加一次费用
        }
        
        if (incrementPointsCost == null) {
            incrementPointsCost = 2; // 默认每增加一个阶梯增加2积分
        }
        
        // 内容长度不足字数阈值，收取基础积分
        if (contentLength <= wordThreshold) {
            return basePointsCost != null ? basePointsCost : 3;
        }
        
        // 超出字数阈值，按阶梯计算
        int incrementCount = (int) Math.ceil((double)(contentLength - wordThreshold) / wordIncrement);
        return (basePointsCost != null ? basePointsCost : 3) + incrementCount * incrementPointsCost;
    }
} 